<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>849</width>
    <height>713</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenGL Template</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QScrollArea" name="scrollArea">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>250</width>
        <height>0</height>
       </size>
      </property>
      <property name="mouseTracking">
       <bool>true</bool>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAsNeeded</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOn</enum>
      </property>
      <property name="sizeAdjustPolicy">
       <enum>QAbstractScrollArea::AdjustIgnored</enum>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>847</width>
         <height>695</height>
        </rect>
       </property>
       <widget class="QLabel" name="param5_label">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>170</y>
          <width>151</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Divergence Angle</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_30">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>300</y>
          <width>16</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Z</string>
        </property>
       </widget>
       <widget class="Line" name="line_9">
        <property name="geometry">
         <rect>
          <x>17</x>
          <y>210</y>
          <width>181</width>
          <height>31</height>
         </rect>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QLabel" name="param2_label">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>50</y>
          <width>161</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Branch Contraction Ratio</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_32">
        <property name="geometry">
         <rect>
          <x>110</x>
          <y>450</y>
          <width>51</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Length</string>
        </property>
       </widget>
       <widget class="QLabel" name="param3_label">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>90</y>
          <width>161</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Branch Angle From Trunk</string>
        </property>
       </widget>
       <widget class="QLabel" name="param4_label">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>130</y>
          <width>151</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Lateral Branching Angle</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_35">
        <property name="geometry">
         <rect>
          <x>60</x>
          <y>270</y>
          <width>151</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Tropism Vector</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_36">
        <property name="geometry">
         <rect>
          <x>90</x>
          <y>300</y>
          <width>16</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Y</string>
        </property>
       </widget>
       <widget class="Line" name="line_10">
        <property name="geometry">
         <rect>
          <x>7</x>
          <y>500</y>
          <width>201</width>
          <height>20</height>
         </rect>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QLabel" name="label_37">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>450</y>
          <width>41</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Width</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_38">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>340</y>
          <width>151</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string> Leaves/Flowers</string>
        </property>
       </widget>
       <widget class="QLabel" name="param1_label">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>10</y>
          <width>151</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Trunk Contraction Ratio</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_40">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>300</y>
          <width>16</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>X</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_42">
        <property name="geometry">
         <rect>
          <x>50</x>
          <y>230</y>
          <width>151</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Tropism Constant</string>
        </property>
       </widget>
       <widget class="Line" name="line_11">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>320</y>
          <width>181</width>
          <height>31</height>
         </rect>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QSlider" name="tropism_constant">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>250</y>
          <width>160</width>
          <height>22</height>
         </rect>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
        <property name="singleStep">
         <number>1</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="Line" name="line_12">
        <property name="geometry">
         <rect>
          <x>8</x>
          <y>650</y>
          <width>201</width>
          <height>20</height>
         </rect>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QCheckBox" name="show_leaves">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>420</y>
          <width>101</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Show Leaves</string>
        </property>
       </widget>
       <widget class="QCheckBox" name="show_flowers">
        <property name="geometry">
         <rect>
          <x>120</x>
          <y>420</y>
          <width>121</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Show Flowers</string>
        </property>
       </widget>
       <widget class="QPushButton" name="export_to_obj">
        <property name="geometry">
         <rect>
          <x>40</x>
          <y>660</y>
          <width>151</width>
          <height>32</height>
         </rect>
        </property>
        <property name="text">
         <string>Export to .obj</string>
        </property>
       </widget>
       <widget class="Line" name="line_13">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>570</y>
          <width>201</width>
          <height>20</height>
         </rect>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QSlider" name="top_branch_radius">
        <property name="geometry">
         <rect>
          <x>8</x>
          <y>530</y>
          <width>171</width>
          <height>22</height>
         </rect>
        </property>
        <property name="value">
         <number>50</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QSlider" name="bottom_branch_raidus">
        <property name="geometry">
         <rect>
          <x>7</x>
          <y>550</y>
          <width>171</width>
          <height>22</height>
         </rect>
        </property>
        <property name="value">
         <number>50</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QLabel" name="label_43">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>510</y>
          <width>151</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>Branch Shape Prameters</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="tropism_x">
        <property name="geometry">
         <rect>
          <x>40</x>
          <y>300</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>0</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="tropism_y">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>300</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>-1</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="tropism_z">
        <property name="geometry">
         <rect>
          <x>160</x>
          <y>300</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>0</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="leaf_x">
        <property name="geometry">
         <rect>
          <x>60</x>
          <y>450</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>.01</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="leaf_z">
        <property name="geometry">
         <rect>
          <x>160</x>
          <y>450</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>.05</string>
        </property>
       </widget>
       <widget class="QLabel" name="tropism_constant_val">
        <property name="geometry">
         <rect>
          <x>180</x>
          <y>250</y>
          <width>31</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>0.0</string>
        </property>
       </widget>
       <widget class="QLabel" name="top_radius_top">
        <property name="geometry">
         <rect>
          <x>188</x>
          <y>530</y>
          <width>31</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>0.0</string>
        </property>
       </widget>
       <widget class="QLabel" name="botto_raidus_val">
        <property name="geometry">
         <rect>
          <x>188</x>
          <y>550</y>
          <width>31</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>0.0</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="param1">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>30</y>
          <width>161</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>.9</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="param2">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>70</y>
          <width>161</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>.6</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="param3">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>110</y>
          <width>161</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>45</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="param4">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>150</y>
          <width>161</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>45</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="param5">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>190</y>
          <width>161</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>137.5</string>
        </property>
       </widget>
       <widget class="QGroupBox" name="groupBox">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>560</y>
          <width>191</width>
          <height>101</height>
         </rect>
        </property>
        <property name="title">
         <string/>
        </property>
        <property name="flat">
         <bool>true</bool>
        </property>
        <widget class="QRadioButton" name="monopodial">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>30</y>
           <width>141</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string comment="0">Monopodial </string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QRadioButton" name="sympodial">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>50</y>
           <width>141</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string comment="0">Sympodial </string>
         </property>
        </widget>
        <widget class="QRadioButton" name="ternary">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>70</y>
           <width>141</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string comment="0">Ternary Branching</string>
         </property>
        </widget>
       </widget>
       <widget class="View" name="view" native="true">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>250</x>
          <y>-10</y>
          <width>640</width>
          <height>700</height>
         </rect>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>250</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>460</y>
          <width>211</width>
          <height>41</height>
         </rect>
        </property>
        <property name="title">
         <string/>
        </property>
        <property name="flat">
         <bool>true</bool>
        </property>
        <property name="checkable">
         <bool>false</bool>
        </property>
        <widget class="QRadioButton" name="maple_leaf">
         <property name="geometry">
          <rect>
           <x>110</x>
           <y>20</y>
           <width>61</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string comment="1">Maple</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="long_leaf">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>10</y>
           <width>71</width>
           <height>41</height>
          </rect>
         </property>
         <property name="text">
          <string comment="1">Round</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
        <zorder>maple_leaf</zorder>
        <zorder>long_leaf</zorder>
        <zorder>long_leaf</zorder>
       </widget>
       <widget class="QCheckBox" name="checkBox_5">
        <property name="geometry">
         <rect>
          <x>40</x>
          <y>370</y>
          <width>141</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Random Placement</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_29">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>390</y>
          <width>191</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Leaves Appear at Thickness:</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="tropism_x_2">
        <property name="geometry">
         <rect>
          <x>190</x>
          <y>390</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>0</string>
        </property>
       </widget>
       <zorder>param5_label</zorder>
       <zorder>label_30</zorder>
       <zorder>line_9</zorder>
       <zorder>param2_label</zorder>
       <zorder>label_32</zorder>
       <zorder>param3_label</zorder>
       <zorder>param4_label</zorder>
       <zorder>label_35</zorder>
       <zorder>label_36</zorder>
       <zorder>line_10</zorder>
       <zorder>label_37</zorder>
       <zorder>label_38</zorder>
       <zorder>param1_label</zorder>
       <zorder>label_40</zorder>
       <zorder>label_42</zorder>
       <zorder>line_11</zorder>
       <zorder>tropism_constant</zorder>
       <zorder>line_12</zorder>
       <zorder>show_leaves</zorder>
       <zorder>show_flowers</zorder>
       <zorder>export_to_obj</zorder>
       <zorder>line_13</zorder>
       <zorder>top_branch_radius</zorder>
       <zorder>bottom_branch_raidus</zorder>
       <zorder>label_43</zorder>
       <zorder>tropism_x</zorder>
       <zorder>tropism_y</zorder>
       <zorder>tropism_z</zorder>
       <zorder>leaf_x</zorder>
       <zorder>leaf_z</zorder>
       <zorder>tropism_constant_val</zorder>
       <zorder>top_radius_top</zorder>
       <zorder>botto_raidus_val</zorder>
       <zorder>param1</zorder>
       <zorder>param2</zorder>
       <zorder>param3</zorder>
       <zorder>param4</zorder>
       <zorder>param5</zorder>
       <zorder>groupBox</zorder>
       <zorder>view</zorder>
       <zorder>view</zorder>
       <zorder>groupBox_3</zorder>
       <zorder>checkBox_5</zorder>
       <zorder>label_29</zorder>
       <zorder>tropism_x_2</zorder>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>View</class>
   <extends>QWidget</extends>
   <header>view.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
